packages/@react-stately/collections/src/Section.ts

```diff
@@ -24,7 +24,7 @@ Section.getCollectionNode = function* getCollectionNode<T>(props: SectionProps<T
     type: 'section',
     props: props,
     hasChildNodes: true,
-    rendered: title,
+    rendered: title || true,
```

Comment:
As it is now, ListLayout won't generate a header if the node doesn't actually render anything: https://github.com/adobe/react-spectrum/blob/cef3714bb8816a4ee64ffb1d0730f2669738d2e4/packages/%40react-stately/layout/src/ListLayout.ts#L313-L319. This is problematic in the RSP Listbox case since we support Sections without titles and thus the above ListLayout change results in a header node not being generated in the set of layout node. However, we don't want to remove the if statement and always provide a header node if a Section is provided since in RAC the user should explicitly provide a Header if they want to render one. Autogenerating a header node if a Section is provided would result in doubling up on Header layout nodes and crashes since the extra node isn't actually a leaf node and doesn't know how to actually render itself.

To compensate, I made the RSP Section's `rendered` always be truthy and added an explicit check that differentiates the `true` case from the user's actual provided title (if any) in ListBoxSection so that we don't render an extranous section header div. Open to any suggestions though

Reply:
seems like this could get tricky because we use
```
{item.rendered &&
```
pretty frequently and now we have to know about this particular special case :(

that said, I don't have a better idea yet, I just don't like this one as a gut reaction

Reply:
yeah, I'm not thrilled with the above change :/